pdm/cli/commands/run.py

```diff
@@ -196,10 +212,46 @@ def _run_task(self, task: Task, args: Sequence[str] = ()) -> int:
             err=True,
             verbosity=termui.DETAIL,
         )
+        if kind == "composite":
+            env = options.get("env", {})
+            env_file = options["env_file"]
+            for script in value:
+                if script is None:
+                    return 1
+                code = self._run_subtask(script, env, env_file)
+                if code != 0:
+                    return code
+            return code
         return self._run_process(
             args, chdir=True, shell=shell, **options  # type: ignore
         )
 
+    def _run_subtask(
+        self, command: str, env: Mapping[str, str], env_file: str | None = None
+    ) -> int:
+        task = self._get_task(command)
+
+        if task is not None:
+            pre_task = self._get_task(f"pre_{command}")
+            if pre_task is not None:
+                code = self._run_task(pre_task.clone_with(env, env_file))
+                if code != 0:
+                    return code
+            code = self._run_task(task.clone_with(env, env_file))
+            if code != 0:
+                return code
+            post_task = self._get_task(f"post_{command}")
+            if post_task is not None:
+                code = self._run_task(post_task.clone_with(env, env_file))
+            return code
+        else:
+            return self._run_process(
```

Comment:
Oh, this is another hidden feature -- we also support string-like `cmd` scripts in composite scripts, so why don't we support all(except for `composite` to avoid recursion) if we are doing so?

Reply:
We now have exact match between tasks and subtask in term of features, including shortcuts because they now use the same function